File: G1_PPAI/G1_PPA1_E1/Entidades/CambioDeEstado.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace G1_PPA1_E1.Entidades
{
    public class CambioDeEstado
    {
        //Atributos
        private DateTime fechaHoraInicio;
        private Estado estado;

        //Constructor
        public CambioDeEstado(DateTime fechaHoraInicio, Estado estado) 
        {
            this.fechaHoraInicio = fechaHoraInicio;
            this.estado = estado;
        }
        //Metodos
        public DateTime getFechaHoraInicio()
        {
            return fechaHoraInicio;
        }
        public Estado getEstado()
        {
            return estado;
        }
        public void esEstadoInicial()
        {

        }
        public void esUltimoEstado()
        {

        }
    }
}

File: PPAI_E3/src/main/java/dsi/g1/entidades/CambioDeEstado.java
package dsi.g1.entidades;

import java.util.Date;

public class CambioDeEstado {

    // atributos
    private Date fechaHoraInicio;
    private Estado estado;

    //constructor
    public CambioDeEstado(Date fechaHoraInicio, Estado estado) {
        this.fechaHoraInicio = fechaHoraInicio;
        this.estado = estado;
    }

    // getters y setters
    public Date getFechaHoraInicio() {
        return fechaHoraInicio;
    }

    public Estado getEstado() {
        return estado;
    }

    public void esEstadoInicial(){}

    public boolean esUltimoEstado(CambioDeEstado cambioAComparar)
    {
        return this.fechaHoraInicio.after(cambioAComparar.getFechaHoraInicio());
    }
}
